src/text-input/input.jsx

```diff
@@ -0,0 +1,269 @@
+      handleAutoresize()
+    }
+    props.hasAutoFocus && inputNode.focus()
+  }, [])
+
+  // instead of componentDidUpdate() method
+  React.useEffect(() => {
+    if (isMultiline) {
+      window.addEventListener("resize", handleAutoresize)
+    } else {
+      window.removeEventListener("resize", handleAutoresize)
+    }
+  }, [isMultiline])
+
+  const handleChange = (event: SyntheticEvent<HTMLInputElement>): void => {
+    const valueFromEvent = event.currentTarget.value
+
+    // Trim value to maxLength if that exists (only on multiline inputs).
+    // Note that this is still required even tho we have the onKeyPress filter
+    // because the user could paste smt in the textarea.
+    const haveToTrim =
+      isMultiline && maxLength && event.currentTarget.value.length > maxLength
+    const inputValue = haveToTrim
+      ? valueFromEvent.substr(0, maxLength)
+      : valueFromEvent
+
+    // propagate to to store and therefore to the input
+    if (onChange) onChange(inputValue, event)
+  }
+
+  const blur = (): void => {
+    inputNode.blur()
+  }
+
+  const focus = (): void => {
+    inputNode.focus()
+  }
+
+  const handleKeyPress = (event: SyntheticEvent<HTMLInputElement>): void => {
+    if (isMultiline && maxLength) {
+      const isReplacing =
+        event.currentTarget.selectionEnd - event.currentTarget.selectionStart
+      const { value: inputValue } = event.currentTarget
+
+      if (!isReplacing && inputValue.length === maxLength) {
+        event.preventDefault()
+        event.stopPropagation()
+      }
+    }
+
+    if (onKeyPress) onKeyPress(event)
+  }
+
+  const handleKeyUp = (
+    event: SyntheticKeyboardEvent<HTMLInputElement>
+  ): void => {
+    type === "password" &&
+      setState({
+        hasCapsWarning: event.getModifierState("CapsLock"),
+      })
+  }
+
+  const isValuePresent = (value: string): boolean =>
+    is(value) && !isEmpty(value)
+
+  const valuePresent = isValuePresent(value) || defaultValue && isValuePresent(defaultValue)
+
+  const inputElementProps:InputElementPropsType = {
+    role,
+    name,
+    defaultValue,
+    disabled: isDisabled,
+    required: isRequired,
+    type,
+    value,
+    placeholder,
+    className: cx(css.inputElement, {
+      [css.filled]: valuePresent,
+    }),
+    ref: inputNode,
+    onKeyUp: handleKeyUp,
+    onKeyPress: handleKeyPress,
+    onChange: handleChange,
+    onFocus,
+    onBlur,
+  }
+
+  if (!isMultiline) {
+    inputElementProps.maxLength = maxLength
+    inputElementProps.onKeyPress = onKeyPress
+  } else {
+    inputElementProps.rows = maxRows
+    inputElementProps.onKeyPress = handleKeyPress
+  }
+
+  const inputClass = cx(
+    css.input,
+    {
+      [css["input--multiline"]]: isMultiline,
+      [css.withBar]: hasBar,
+      [css.disabled]: isDisabled,
+      [css.errored]: error,
+      [css.hidden]: type === "hidden",
+      [css.withIcon]: icon,
+      [inputClassName || ""]: is(inputClassName),
+    },
+    className
```

Comment:
Should this line be removed? The `className` class is being used in the input wrapper already.